From fbdb26606e37f78e99623476524d0fe88e17dd6f Mon Sep 17 00:00:00 2001
From: Voon Weifeng <weifeng.voon@intel.com>
Date: Sat, 27 Jul 2019 07:49:49 +0800
Subject: [PATCH 2/4] net: stmmac: add EHL PSE0 & PSE1 2.5Gbps PCI info and PCI
 ID

Add EHL PSE0/1 SGMII 2.5Gbps PCI info and PCI ID
Port from Yocto mainline:
https://github.com/intel/linux-intel-lts/commit/0554685b9c04d19e2468d17090591861bb88a5cd

Tracked-On: OAM-99178
Signed-off-by: Voon Weifeng <weifeng.voon@intel.com>
Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
Change-Id: Ide2fef40aa8c024154cfbb7e557846d021837c69
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
index 77bd844f4975..81330453fd0a 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
@@ -648,8 +648,10 @@ static SIMPLE_DEV_PM_OPS(stmmac_pm_ops, stmmac_pci_suspend, stmmac_pci_resume);
 #define STMMAC_EHL_SGMII1G_ID	0x4b31
 #define STMMAC_EHL_PSE0_RGMII1G_ID	0x4ba0
 #define STMMAC_EHL_PSE0_SGMII1G_ID	0x4ba1
+#define STMMAC_EHL_PSE0_SGMII2G5_ID	0x4ba2
 #define STMMAC_EHL_PSE1_RGMII1G_ID	0x4bb0
 #define STMMAC_EHL_PSE1_SGMII1G_ID	0x4bb1
+#define STMMAC_EHL_PSE1_SGMII2G5_ID	0x4bb2
 #define STMMAC_TGL_SGMII1G_ID	0xa0ac
 #define STMMAC_GMAC5_ID		0x7102

@@ -668,10 +670,14 @@ static const struct pci_device_id stmmac_id_table[] = {
 		      ehl_pse0_rgmii1g_pci_info),
 	STMMAC_DEVICE(INTEL, STMMAC_EHL_PSE0_SGMII1G_ID,
 		      ehl_pse0_sgmii1g_pci_info),
+	STMMAC_DEVICE(INTEL, STMMAC_EHL_PSE0_SGMII2G5_ID,
+		      ehl_pse0_sgmii1g_pci_info),
 	STMMAC_DEVICE(INTEL, STMMAC_EHL_PSE1_RGMII1G_ID,
 		      ehl_pse1_rgmii1g_pci_info),
 	STMMAC_DEVICE(INTEL, STMMAC_EHL_PSE1_SGMII1G_ID,
 		      ehl_pse1_sgmii1g_pci_info),
+	STMMAC_DEVICE(INTEL, STMMAC_EHL_PSE1_SGMII2G5_ID,
+		      ehl_pse1_sgmii1g_pci_info),
 	STMMAC_DEVICE(INTEL, STMMAC_TGL_SGMII1G_ID, tgl_sgmii1g_pci_info),
 	STMMAC_DEVICE(SYNOPSYS, STMMAC_GMAC5_ID, snps_gmac5_pci_info),
 	{}
--
2.17.1

